tests/gtest/aviflosslesstest.cc

```diff
@@ -0,0 +1,95 @@
+       {"paris_icc_exif_xmp.png", "paris_exif_xmp_icc.jpg"}) {
+    // Read a ground truth image with default matrix coefficients.
+    const std::string file_path = std::string(data_path) + file_name;
+    const testutil::AvifImagePtr ground_truth_image =
+        testutil::ReadImage(data_path, file_name);
+
+    for (auto matrix_coefficient :
+         {AVIF_MATRIX_COEFFICIENTS_IDENTITY, AVIF_MATRIX_COEFFICIENTS_YCGCO}) {
+      // Read a ground truth image but ask for certain matrix coefficients.
+      testutil::AvifImagePtr image(avifImageCreateEmpty(), avifImageDestroy);
+      ASSERT_NE(image, nullptr);
+      image->matrixCoefficients = (avifMatrixCoefficients)matrix_coefficient;
+      const avifAppFileFormat file_format = avifReadImage(
+          file_path.c_str(),
+          /*requestedFormat=*/AVIF_PIXEL_FORMAT_NONE,
+          /*requestedDepth=*/0,
+          /*chromaDownsampling=*/AVIF_CHROMA_DOWNSAMPLING_AUTOMATIC,
+          /*ignoreICC=*/false, /*ignoreExif=*/false,
+          /*ignoreXMP=*/false, image.get(),
+          /*outDepth=*/nullptr, /*sourceTiming=*/nullptr,
+          /*frameIter=*/nullptr);
+      ASSERT_NE(file_format, AVIF_APP_FILE_FORMAT_UNKNOWN);
+
+      // Encode.
+      testutil::AvifEncoderPtr encoder(avifEncoderCreate(), avifEncoderDestroy);
+      ASSERT_NE(encoder, nullptr);
+      encoder->speed = AVIF_SPEED_FASTEST;
+      encoder->quality = AVIF_QUALITY_LOSSLESS;
+      encoder->qualityAlpha = AVIF_QUALITY_LOSSLESS;
+      testutil::AvifRwData encoded;
+      avifResult result =
+          avifEncoderWrite(encoder.get(), image.get(), &encoded);
+      ASSERT_EQ(result, AVIF_RESULT_OK) << avifResultToString(result);
+
+      // Decode to RAM.
+      testutil::AvifImagePtr decoded(avifImageCreateEmpty(), avifImageDestroy);
+      ASSERT_NE(decoded, nullptr);
+      testutil::AvifDecoderPtr decoder(avifDecoderCreate(), avifDecoderDestroy);
+      ASSERT_NE(decoder, nullptr);
+      result = avifDecoderReadMemory(decoder.get(), decoded.get(), encoded.data,
+                                     encoded.size);
+      ASSERT_EQ(result, AVIF_RESULT_OK) << avifResultToString(result);
+
+      // Convert to a temporary PNG and read back, to have default matrix
+      // coefficients.
+      const std::string temp_dir = testing::TempDir();
+      const std::string temp_file = "decoded_default.png";
+      const std::string tmp_path = temp_dir + temp_file;
+      ASSERT_TRUE(testutil::WriteImage(decoded.get(), tmp_path.c_str()));
+      const testutil::AvifImagePtr decoded_default =
+          testutil::ReadImage(temp_dir.c_str(), temp_file.c_str());
+
+      // Verify that the ground truth and decoded images are the same.
+      const bool are_images_equal =
+          testutil::AreImagesEqual(*ground_truth_image, *decoded_default);
+      if (matrix_coefficient == AVIF_MATRIX_COEFFICIENTS_IDENTITY) {
+        ASSERT_TRUE(are_images_equal);
+      } else {
+        ASSERT_FALSE(are_images_equal);
```

Comment:
This is not lossless because `AVIF_MATRIX_COEFFICIENTS_YCGCO` uses only 8 bits, correct?

Reply:
Exactly, let me add a comment.